packages/medusa/src/api-v2/store/payment-collections/[id]/payment-sessions/route.ts

```diff
@@ -0,0 +1,64 @@
+import { createPaymentSessionsWorkflow } from "@medusajs/core-flows"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPaymentModuleService } from "@medusajs/types"
+import { remoteQueryObjectFromString } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+import { defaultStorePaymentCollectionFields } from "./query-config"
+import { StorePostPaymentCollectionsPaymentSessionReq } from "./validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<StorePostPaymentCollectionsPaymentSessionReq>,
+  res: MedusaResponse
+) => {
+  const { id } = req.params
+  const { context, provider_id, data } = req.body
+
+  // If the customer is logged in, we auto-assign them to the payment collection
+  if (req.auth?.actor_id) {
+    context.customer = {
+      ...context.customer,
+      id: req.auth.actor_id,
+    }
+  }
+
+  const paymentService = req.scope.resolve<IPaymentModuleService>(
+    ModuleRegistrationName.PAYMENT
+  )
+
+  const paymentCollection = await paymentService.retrievePaymentCollection(id, {
```

Comment:
Should we use remoteQuery instead? Alternatively, should the data fetching happen in the workflow itself?

Reply:
I decided to use the service directly, since we don't need any resources outside of the payment module's domain. We could use the `remoteQuery` for consistency.

That said, I think I'll move the retrieval to the workflow for now, since that would eliminate this concern.

Reply:
Sounds good 👍